scipy/optimize/tests/test_minpack.py

```diff
@@ -394,6 +394,13 @@ def test_concurrent_with_gradient(self):
 
 
 class TestCurveFit(object):
+    def test_empty_array_with_bounds(self):
+            # This test is regarding the issue #9864
+            # Test that Curve_fit checks for empty input data
+            # if called with bounds
+            # Calling this function will produce a meaningful error message.
+            curve_fit(lambda x, a: a*x, [], [], bounds = (0, 1))
+
```

Comment:
I guess you need to use `assert_raises` here, otherwise the tests fail due to the (expected) error.

Reply:
Ohk, let me try that.

Reply:
Yup, you were right. It passed all the tests. I hope it can be merged now. Thank you for your help.